piptools/writer.py

```diff
@@ -258,15 +277,14 @@ def _format_requirement(
             required_by.add(_comes_from_as_string(ireq))
 
         if required_by:
-            sorted_required_by = sorted(required_by)
-            if len(sorted_required_by) == 1:
-                source = sorted_required_by[0]
-                annotation = "    # via " + source
-            else:
-                annotation_lines = ["    # via"]
-                for source in sorted_required_by:
-                    annotation_lines.append("    #   " + source)
-                annotation = "\n".join(annotation_lines)
-            line = f"{line}\n{comment(annotation)}"
+            annotation = {
+                "split": annotation_style_split,
+                "line": annotation_style_line,
+            }[self.annotation_style](required_by)
+            sep = {"split": "\n    ", "line": "\n    " if ireq_hashes else "  "}[
+                self.annotation_style
+            ]
+            lines = f"{line:24}{sep}{comment(annotation)}".splitlines()
```

Comment:
What does the magic number 24 mean? Maybe add a comment or give it a name to make it self-documenting.

Reply:
I took it from the previous implementation. I don't know if it's derived from anything special, so I have no knowledge to add.

Reply:
The new and third commit adds a comment that aims to convey a reassuring shrug in more words.

Reply:
nice 👍🏽 comments are usually before the line they apply to, though, unless pip-tools uses an unusual convention for this.

Reply:
Thanks, swapped the lines